keys/keys_test.go

```diff
@@ -75,6 +76,25 @@ func TestKeyAddress(t *testing.T) {
 	}
 }
 
+func TestMakeTableIndexKey(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	key := MakeTableIndexKey(12, 345, []byte("foo"), []byte("bar"))
+	expKey := MakeKey(TableDataPrefix, encoding.EncodeUvarint(nil, 12), encoding.EncodeUvarint(nil, 345), encoding.EncodeBytes(nil, []byte("foo")), encoding.EncodeBytes(nil, []byte("bar")))
```

Comment:
This is just testing that `MakeTableIndexKey` has a particular implementation. What I'd rather see is something which generated a few index keys using different table, index and column values and verified they sort correctly. For example, something like:

```
  keys := []proto.Key{
    MakeTableIndexKey(0, 0, []byte("foo")),
    MakeTableIndexKey(0, 0, []byte("fooo")),
    MakeTableIndexKey(0, 1, []byte("bar")),
    MakeTableIndexKey(1, 0, []byte("bar")),
    MakeTableIndexKey(1, 0, []byte("bar"), []byte("foo")),
  }
  for i := 1; i < len(keys); i++ {
    if keys[i-1] >= keys[i] {
      t.Errorf("key %d >= key %d", i - 1, i)
    }
  }
```